spec.md

```diff
@@ -1,94 +1,120 @@
 
-On receiving a request, the request should be checked:
+On receiving a request, the following must be checked:
 
 - The DNA hash string must decode to a valid DNA hash.
-- The coordinator identifier and zome name must be valid UTF-8 and at most 100 characters long.
-- The function name must be a valid function identifier (valid UTF-8 is good enough for now) and at most 100 characters long.
+- The coordinator identifier, zome name and function name must be valid UTF-8 and at most 100 characters long.
 - The function name must be included in the list of allowed functions for the app.
 - The payload length must be within the configured limit.
-- The payload must be valid base64 url and decode to valid JSON.
+- The payload must be valid Base64 url encoded and decode to valid JSON.
 
 ### Identify the app to call
 
-The gateway uses a single admin websocket to make requests to Holochain. At this point, either use the existing connection or open a new one if the websocket is closed.
+The gateway should use a single admin websocket to make requests to Holochain. Either use the existing connection or 
+open a new one if the websocket is closed.
 
-When receiving a valid request, the gateway first checks its cached list of installed apps. If the cache is empty or the app isn't found in the cache, it calls the `ListApps` operation on Holochain with the filter set to only return running apps. The response is stored in the cache for future requests.
+When receiving a valid request, the gateway should first check its cached list of installed apps. If the app isn't 
+found in the cache, call the `ListApps` operation on Holochain with the filter set to only return running apps. The 
+response should be stored in the cache for future requests.
 
-The gateway identifies apps that match both the DNA hash specified in the request and have an installed app ID matching the `coordinator-identifier`. It then checks if the app ID is in the list of allowed apps configured in `AllowedAppIds`. If multiple matching apps are found, an error is returned since a unique app cannot be determined.
+The gateway must identify apps that match both the DNA hash specified in the request and have an installed app ID
+matching the `coordinator-identifier`. It must then check if the app ID is in the list of allowed apps configured in 
+`AllowedAppIds`. If multiple matching apps are found, an error should be returned since a unique app cannot be 
+determined.
 
-If no app is found in the initial check but the cache was populated, a fresh request to `ListApps` is made to update the cache, and the search is performed again. If the app still isn't found, or if it's found but not in the allowed list, an appropriate error is returned.
+If no app was found in the initial check but the cache was repopulated, the search is performed again. If the app still 
+isn't found, or if it's found but not in the allowed list, an appropriate error must be returned.
 
-### Prepare zome call signing
```

Comment:
This section isn't really relevant after implementation. Preparing zome call singing was just about preparing signing and that is now done when opening app websocket connections so it should be merged into that section